src/styles/main.scss

```
@@ -1,3 +1,53 @@
-body {
-  margin: 0;
+.calendar {
+  width: 707px;
+  margin: 0 auto;
+  display: flex;
+  flex-wrap: wrap;
+  gap: 1px;
+  padding: 30px 10px 10px;
+
+  $days: mon, tue, wed, thu, fri, sat, sun;
+
+  @each $day in $days {
+    &--start-day-#{$day} :first-child {
+      margin-left: (index($days, $day) - 1) * 101px;
+    }
+  }
+
+  $month_lengths: 28, 29, 30, 31;
+
+  @each $length in $month_lengths {
+    &--month-length--#{$length} &__day:nth-child(n + #{$length + 1}) {
+      display: none;
+    }
+  }
+
+  $days: 31;
+
+  @for $i from 1 through $days {
+    &__day:nth-child(#{$i}):before {
+      content: "#{$i}";
+      font-family: Arial, sans-serif;
+      font-size: 30px;
+    }
+  }
+
+  &__day {
+    box-sizing: border-box;
+    background-color: #eee;
+    width: 100px;
+    height: 100px;
+    border: 1px solid #000;
+    display: flex;
+    justify-content: center;
+    align-items: center;
+    transition: transform 0.5s;
+
+    &:first-child:hover,
+    &:nth-child(n +2):hover {
```

I don't get those selectors, it's overcomplicated. Instead just do:
```suggestion
&:hover {
```